### ISSUE TITLE
Issue in Results--Object Detection API eval.py

### ISSUE THREAD

**Issue by rabiaali95** [tensorflow/models#6539]
I have 203 test images and 3 classes. I have modified my config file for ssd_mobilenet_v1_coco accordingly. Now when i run eval.py, all test images are evaluated twice and i see results twice. Also i can not see all metrics like average recall and precision at different IoUs.
![img](https://user-images.githubusercontent.com/31028173/55693574-b7a87280-59ea-11e9-842b-de2e89b3be6e.png)

Does anyone know how to solve this issue?

Here is the modifies config file

model {
  ssd {
    num_classes: 3
    box_coder {
      faster_rcnn_box_coder {
        y_scale: 10.0
        x_scale: 10.0
        height_scale: 5.0
        width_scale: 5.0
      }
    }
    matcher {
      argmax_matcher {
        matched_threshold: 0.5
        unmatched_threshold: 0.5
        ignore_thresholds: false
        negatives_lower_than_unmatched: true
        force_match_for_each_row: true
      }
    }
    similarity_calculator {
      iou_similarity {
      }
    }
    anchor_generator {
      ssd_anchor_generator {
        num_layers: 6
        min_scale: 0.2
        max_scale: 0.95
        aspect_ratios: 1.0
        aspect_ratios: 2.0
        aspect_ratios: 0.5
        aspect_ratios: 3.0
        aspect_ratios: 0.3333
      }
    }
    image_resizer {
      fixed_shape_resizer {
        height: 300
        width: 300
      }
    }
    box_predictor {
      convolutional_box_predictor {
        min_depth: 0
        max_depth: 0
        num_layers_before_predictor: 0
        use_dropout: false
        dropout_keep_probability: 0.8
        kernel_size: 1
        box_code_size: 4
        apply_sigmoid_to_scores: false
        conv_hyperparams {
          activation: RELU_6,
          regularizer {
            l2_regularizer {
              weight: 0.00004
            }
          }
          initializer {
            truncated_normal_initializer {
              stddev: 0.03
              mean: 0.0
            }
          }
          batch_norm {
            train: true,
            scale: true,
            center: true,
            decay: 0.9997,
            epsilon: 0.001,
          }
        }
      }
    }
    feature_extractor {
      type: 'ssd_mobilenet_v1'
      min_depth: 16
      depth_multiplier: 1.0
      conv_hyperparams {
        activation: RELU_6,
        regularizer {
          l2_regularizer {
            weight: 0.00004
          }
        }
        initializer {
          truncated_normal_initializer {
            stddev: 0.03
            mean: 0.0
          }
        }
        batch_norm {
          train: true,
          scale: true,
          center: true,
          decay: 0.9997,
          epsilon: 0.001,
        }
      }
    }
    loss {
      classification_loss {
        weighted_sigmoid {
        }
      }
      localization_loss {
        weighted_smooth_l1 {
        }
      }
      hard_example_miner {
        num_hard_examples: 3000
        iou_threshold: 0.99
        loss_type: CLASSIFICATION
        max_negatives_per_positive: 3
        min_negatives_per_image: 0
      }
      classification_weight: 1.0
      localization_weight: 1.0
    }
    normalize_loss_by_num_matches: true
    post_processing {
      batch_non_max_suppression {
        score_threshold: 1e-8
        iou_threshold: 0.6
        max_detections_per_class: 100
        max_total_detections: 100
      }
      score_converter: SIGMOID
    }
  }
}

train_config: {
  batch_size: 10
  optimizer {
    rms_prop_optimizer: {
      learning_rate: {
        exponential_decay_learning_rate {
          initial_learning_rate: 0.004
          decay_steps: 800720
          decay_factor: 0.95
        }
      }
      momentum_optimizer_value: 0.9
      decay: 0.9
      epsilon: 1.0
    }
  }
  fine_tune_checkpoint: "ssd_mobilenet_v1_coco_11_06_2017/model.ckpt"
  from_detection_checkpoint: true
  num_steps: 200000
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
  data_augmentation_options {
    ssd_random_crop {
    }
  }
}

train_input_reader: {
  tf_record_input_reader {
    input_path: "data/train.record"
  }
  label_map_path: "data/object-detection.pbtxt"
}

eval_config: {
  num_examples: 203
  num_visualizations: 203
  max_evals: 10
}

eval_input_reader: {
  tf_record_input_reader {
    input_path: "data/test.record"
  }
  label_map_path: "data/object-detection.pbtxt"
  shuffle: false
  num_readers: 1
}
What is the top-level directory of the model you are using: models-master/research/object_detection
Have I written custom code: No (The changes in config file are as above)
OS Platform and Distribution: Ubuntu 16.04
TensorFlow installed from: GPU version of Tensorflow
TensorFlow version: 1.12.0
Bazel version: NA: don't know what is Bazel
CUDA/cuDNN version: NA (not the related to the problem)
GPU model and memory: Titan V
Exact command to reproduce: 
    python3  eval.py \
    --logtostderr \
    --checkpoint_dir=training \
    --eval_dir=eval \
    --pipeline_config_path=training/ssd_mobilenet_v1_coco.config

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by adifdwimaulana** [comment]
Have you found the solution for this issue ?

**Comment by ibrahim1611** [comment]
I am facing same problem. I want to test 26 images using eval.py but on tensorboard it shows only 9 images. Also the graphs just show a single point. Can anyone please let me know how to solve this problem.

**Comment by adifdwimaulana** [comment]
> I am facing same problem. I want to test 26 images using eval.py but on tensorboard it shows only 9 images. Also the graphs just show a single point. Can anyone please let me know how to solve this problem.

You have to set num_visualization parameter in eval_config. So your eval_config should be :
eval_config: {
  num_examples: 26
  num_visualizations: 26
  max_evals: 10
}
